src/store/messagesStore.js

```diff
@@ -246,7 +246,7 @@ const getters = {
 
 	// Returns true if the message has reactions
 	hasReactions: (state) => (token, messageId) => {
-		return Object.keys(state.messages[token][messageId].reactions).length !== 0
+		return Object.keys(Object(state.messages[token]?.[messageId]?.reactions)).length !== 0
```

Comment:
It could be either a room token or message, but from what I can see, in Message.vue we have a computed property, which is called for EACH component. so with high chances it could be an undefined, and therefore field 'reactions' could not be found. So it's fine by me to have only a safety check here

Reply:
> It could be either a room token or message, but from what I can see, in Message.vue we have a computed property, which is called for EACH component. so with high chances it could be an undefined, and therefore field 'reactions' could not be found.

Is there a case when `Message` component receives `token` or `id` of a message or conversation we do not have in the store?

Reply:
Can't name a specific case, but we're deleting messages from the store from time to time:
- when sending, delete temporary messages
- when switching between rooms, ease message list
- when perform a full conversation list update, also could be
- when manually delete message
- when dispatch `deleteConversation` :warn: or `clearConversationHistory` actions (first one is most likely the culprit)
Don't know, how to reproduce it propoerly, but it sometimes happen, when you leave the page open => which leads us to conversation fetching again

Reply:
> So it's fine by me to have only a safety check here

Coming back to my previous answer: almost for all getters in `messagesStore` we have a simple check for data existence:
```js
if (state.messages[token]) {
    return Object.values(state.messages[token])
}
```

In case of `hasReactions` getters it's missing, so I'd consider this change as the same check for existence